internal/runbits/buildlogfile/buildlogfile.go

```diff
@@ -0,0 +1,245 @@
+package buildlogfile
```

Comment:
This doesn't feel like it should be inside a runbit, or deal with output. Are we doing this due to the way event handling is set up? 

I would expect to see the majority of this logic somewhere inside the runtime package dir (some sub-package). The output part should be entirely separate from this.

Reply:
Seeing the rest of the PR, it's clearer to me that in fact a lot of this logic DOES live in the runtime package. Which now makes me wonder; why is there so much logic in this runbit? It's not just passing data along, it's doing requests to get log files from the server. It feels a little unclear to me how the responsibilities of these packages is determined.

Reply:
I think that is a valuable criticism:  The download of the log files on failures, doesn't need to happen here.  That should probably move up the chain, and I think that the `consumer` would be the best place to handle that.

Reply:
I actually put it in the `producer` because it is producing events after downloading the artifact log.  That seems much cleaner to me.